### ISSUE TITLE
Move Gears data directory on mac, and handle data migration

### ISSUE THREAD

**Issue by GoogleCodeExporter** [edrikL/gears#541]
```
Currently the Gears data directory on mac firefox (not sure about safari) 
is in ~/Library/Caches/.... This directory is volatile and we are not 
supported to store anything of importance there:

> Bug report from apple:
>
> This is a courtesy email regarding Bug ID# 5981589.
> ...
> 3) ~/Library/Caches and /Library/Caches are only supposed to be used
> to store information which can be re-created on demand at any time if
> needed.  It should be perfectly safe to blow these directories away at
> any time and the only thing the user should notice is some decreased
> performance as the cache is rebuilt. If there are applications storing
> data there which is not really a "cache" then those applications are
> violating Mac OS X's programming guidelines and bugs should be written
> against them.

The reason we are using this location is because we find our data directory 
by using nsIDirectoryService with the key "ProfLD". This key makes sense on 
Windows because it maps to a directory in ~\Local Settings\..., which is 
what we want.

Basically, ProfD and ProfLD do not mean exactly the same things on Windows 
and Mac. On Mac, we should use ProfD, and on Windows we should use ProfLD.

Hooray!

For more, see this thread:
http://groups.google.com/group/gears-
users/browse_thread/thread/1b5fc0396704c4e2

```

Original issue reported on code.google.com by `zbo...@gmail.com` on 13 Jun 2008 at 1:23
